### ISSUE TITLE
Dynamic badge generator generates wrong URL

### ISSUE THREAD

**Issue by valeth** [badges/shields#1339]
When I tried to use the dynamic badge creator [here](http://shields.io/#your-badge) the inputted URL in the uri field generated a wrong URL containing two dashes (some--path) instead of one (some-path).

Manually correcting the URL fixes this.

The parameters used were:
```
type:      json
label:     UserScript
uri:       https://gitlab.com/valeth/wanikani-lesson-cap/raw/master/package.json
path:      $.version
hex color: <default>
prefix:    v
suffix:    <default>
```

Result:
![badge](https://img.shields.io/badge/dynamic/json.svg?label=UserScript&colorB=&prefix=v&suffix=&query=$.version&uri=https%3A%2F%2Fgitlab.com%2Fvaleth%2Fwanikani--lesson--cap%2Fraw%2Fmaster%2Fpackage.json)
`https://img.shields.io/badge/dynamic/json.svg?label=UserScript&colorB=&prefix=v&suffix=&query=$.version&uri=https%3A%2F%2Fgitlab.com%2Fvaleth%2Fwanikani--lesson--cap%2Fraw%2Fmaster%2Fpackage.json`

Fixed (and working) version:
![badge](https://img.shields.io/badge/dynamic/json.svg?label=UserScript&colorB=&prefix=v&suffix=&query=$.version&uri=https%3A%2F%2Fgitlab.com%2Fvaleth%2Fwanikani-lesson-cap%2Fraw%2Fmaster%2Fpackage.json)
`https://img.shields.io/badge/dynamic/json.svg?label=UserScript&colorB=&prefix=v&suffix=&query=$.version&uri=https%3A%2F%2Fgitlab.com%2Fvaleth%2Fwanikani-lesson-cap%2Fraw%2Fmaster%2Fpackage.json`

**Comment by tooomm** [comment]
Same problem when I try to access https://mtgjson.com/json/version-full.json (`$.version`).
It returns: https://img.shields.io/badge/dynamic/json.svg?label=&colorB=&prefix=&suffix=&query=$.version&uri=https%3A%2F%2Fmtgjson.com%2Fjson%2Fversion--full.json with `--` at the end.

**Comment by paulmelnikow** [comment]
Is this fixed in the new frontend on staging?

https://shields-staging.herokuapp.com

**Comment by tooomm** [comment]
Well, it looks to be completely broken there (waited for full load). 😆 
The `Make Badge` button won't work at all and in the `type` field at the beginning there is always "json" displayed. You also can't remove that. :)

**Comment by paulmelnikow** [comment]
Ah, yes, on staging the URL is missing `/badge` before `/dynamic`. Can you confirm it's working otherwise?

We could remove that **json** field, and change **url** to **json url**. It was done that way it is because @RedSparr0w wants to add a dynamic XML badge.

**Comment by tooomm** [comment]
>We could remove that json field, and change url to json url. It was done that way it is because @RedSparr0w wants to add a dynamic XML badge.

Oh, I thought xml is working already. I really like additional options like `xml` or `yml/yaml`!

>Ah, yes, on staging the URL is missing /badge before /dynamic. Can you confirm it's working otherwise?

Looking at the code I realized that the button gets only enabled if you fill out certain information...
After clicking it and adding in `/badge` it works!

**Comment by paulmelnikow** [comment]
Proposed fix: #1344
